src/block_gmres.jl

```diff
@@ -0,0 +1,349 @@
+# An implementation of block-GMRES for the solution of the square linear system AX = B.
+#
+# Alexis Montoison, <alexis.montoison@polymtl.ca>
+# Chicago, October 2023.
+
+include("block_utils.jl")
+
+export BlockGmresSolver, block_gmres, block_gmres!
+
+"""
+Type for storing the vectors required by the in-place version of BLOCK-GMRES.
+
+The outer constructors
+
+    solver = BlockGmresSolver(m, n, p, memory, S)
+    solver = BlockGmresSolver(A, B, memory=5)
+
+may be used in order to create these vectors.
+`memory` is set to `div(n,p)` if the value given is larger than `div(n,p)`.
+"""
+mutable struct BlockGmresSolver{T,FC,S} <: KrylovSolver{T,FC,S}
```

Comment:
I would suggest renaming as `BlockGMRESSolver`

Reply:
I use the following notation in Krylov.jl: https://jso.dev/Krylov.jl/dev/inplace/

I could change the notation for block-Krylov methods.